clang/include/clang/Basic/TargetInfo.h

```diff
@@ -1114,10 +1114,12 @@ class TargetInfo : public TransferrableTargetInfo,
 
     std::string ConstraintStr;  // constraint: "=rm"
     std::string Name;           // Operand name: [foo] with no []'s.
+    unsigned FlagOutputCCUpperBound;
```

Comment:
I am wondering if we can re-use the existing range fields `ImmRange` here.  Those are currently only used for input operands and require that only immediates within this range are used as input.   For an output operand, this isn't useful - but instead we could take it to mean that the output is logically constrained to fall within that range, so we can add appropriate assertions.